src/main/java/se/sundsvall/supportmanagement/integration/db/util/ErrandNumberGeneratorService.java

```diff
@@ -0,0 +1,47 @@
+package se.sundsvall.supportmanagement.integration.db.util;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import org.springframework.stereotype.Component;
+
+import se.sundsvall.supportmanagement.integration.db.ErrandNumberSequenceRepository;
+import se.sundsvall.supportmanagement.integration.db.model.ErrandNumberSequenceEntity;
+
+@Component
+public class ErrandNumberGeneratorService {
+
+	private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyMM");
+
+	private final ErrandNumberSequenceRepository repository;
+
+	public ErrandNumberGeneratorService(final ErrandNumberSequenceRepository errandNumberSequenceRepository) {
+		this.repository = errandNumberSequenceRepository;
+	}
+
+	public String generateErrandNumber(final String namespace) {
```

Comment:
Do we need a transactional-annotation here with a restirctive isolation level?

In order to prevent scenarios with concurrent transactions generating (identical errand numbers)?

Reply:
To be honest I have no real insight in the work load of this service but better safe than sorry so @Transactional it is.